### PR TITLE
[INF] Make requirements.txt smaller #299

### LINKED ISSUE #299: [INF] Make requirements.txt smaller
Follow-up from #257 

The idea is to have feature-specific requirements.txt. Such as for biology, specifically requires biopython.

so we can install the package per feature as needed, such as with extra biology. It goes `pip install "pyjanitor[biology]"`

The example of such implementations in `setup.py` is available at this link: https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies

### PR DESCRIPTION
**Pull request by eyaltrabelsi**
# PR Description

Please describe the changes proposed in the pull request: 

-allow feature-specific requirements.txt. Such as for biology, specifically requires biopython.

**This PR resolves #299.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:master, but rather from `<your_username>`:<branch_name>.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [ ] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build
    
Once done, please check off the check-box above.
    
If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [ ] Check to ensure that test coverage covers the lines of code that you have added.
    - [ ] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [x] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,6 +52,11 @@ def generate_long_description() -> str:
     url="https://github.com/ericmjl/pyjanitor",
     packages=["janitor"],
     install_requires=requirements(),
+    extras_require={
+            'all': extra_all,
+            'biology': extra_biology,
+            'chemistry': extra_chemistry,
+            'engineering': extra_engineering}
```
Comment (ericmjl) [inline setup.py]: Missing comma is causing build to fail.

```suggestion
            'engineering': extra_engineering},
```

Comment (eyaltra) [reply setup.py]: to be honest i dont know, I will need to check both

Comment (ericmjl) [reply setup.py]: No worries.

If you want, you can use the "commit suggestion" functionality to accept the suggestion directly without needing to touch your text editor :smile:.

**Comment by zjpoh** [comment]
Should we also add pyspark as an option?

**Comment by ericmjl** [comment]
@zjpoh good catch! @eyaltrabelsi, could you make the change please?

**Comment by eyaltrabelsi** [comment]
rdkit seems to be ["not pip installable"](https://github.com/rdkit/rdkit/issues/1812) only condas, @ericmjl any recommendation ?

**Comment by ericmjl** [comment]
@eyaltrabelsi let’s deop the chemistry option for now. RDKit is a bit of a pain to install without Conda.

File: setup.py [hunk 2]
```diff
@@ -38,6 +38,12 @@ def generate_long_description() -> str:
     return long_description
 
 
+extra_spark = ["pyspark"]
+extra_biology = ["biopython"]
+extra_chemistry = ["rdkit"]
+extra_engineering = ["unyt"]
+extra_all = extra_biology + extra_chemistry + extra_engineering + extra_spark
```
Comment (ericmjl) [inline setup.py]: I think `extra_chemistry` will cause it to fail if someone installs `pyjanitor[all]`. Could we comment it out until later?

Comment (eyaltra) [reply setup.py]: @ericmjl fixed :)

**Comment by ericmjl** [comment]
Thanks @eyaltrabelsi!

**Pull request merged**